app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/UserServiceCEImpl.java

```diff
@@ -659,16 +659,31 @@ public Flux<User> get(MultiValueMap<String, String> params) {
         return Flux.error(new AppsmithException(AppsmithError.UNSUPPORTED_OPERATION));
     }
 
+    private Boolean validateName(String name){
+        /*
+            Regex allows for Accented characters and alphanumeric with some special characters dot (.), apostrophe ('),
+            hyphen (-) and spaces
+         */
+        Boolean isValidName = name.matches("^[A-Za-zÀ-ÖØ-öø-ÿ0-9 .'-]+$");
+        return isValidName;
+    }
```

Comment:
This will never result in a `null`, so we can prefer to use the non-boxed version `boolean`.

```suggestion
    private boolean validateName(String name){
        /*
            Regex allows for Accented characters and alphanumeric with some special characters dot (.), apostrophe ('),
            hyphen (-) and spaces
         */
        boolean isValidName = name.matches("^[A-Za-zÀ-ÖØ-öø-ÿ0-9 .'-]+$");
        return isValidName;
    }
```

Reply:
Also, let's move this regex pattern to a static constant, not as a `String`, but a compiled `Pattern`, so the JDK doesn't have to compile the regex on every function call.

Example: https://github.com/appsmithorg/appsmith/blob/439ceb54064aec4c7170ae701c28e1e2267fefb4/app/server/appsmith-git/src/main/java/com/appsmith/git/helpers/FileUtilsImpl.java#L82